packages/haunted-hooks-testing-library/test/main.ts

```diff
@@ -8,44 +8,374 @@ import {
 } from "haunted";
 import { act, renderHook } from "../src";
 
-// BUG issues with props and stuff
+describe("Async Utils", () => {
+  describe("waitForNextUpdate", () => {
+    test("default", async () => {
+      function hook() {
+        const [counter, setCounter] = useState(0);
+
+        setTimeout(() => setCounter((c = 0) => c + 1), 10);
```

Comment:
This is a more idiomatic way of writing this to avoid it running infinitely

```suggestion
        useEffect(()=>{
        const fn = () => setCounter((c = 0) => c + 1)
        setInterval(fn, 10);
        return ()=>clearInteval(fn);
     }, []);
```

Reply:
Done. I'll do the same thing in Universal Hooks Testing Library.